third_party/meshroom/meshroom/nodes/ARTwin/HololensPointcloudComposer.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import print_function
+
+__version__ = "0.1"
+
+from meshroom.core import desc
+import shutil
+import glob
+import os
+import sys
+os.environ["KMP_DUPLICATE_LIB_OK"]="TRUE"
+
+# import mapper packages
+dir_path = __file__
+for i in range(6):
+    dir_path = os.path.dirname(dir_path)
+sys.path.append(dir_path)
+from src.holo.HoloIO import HoloIO
+
+class HololensPointcloudComposer(desc.Node):
+
+    category = 'ARTwin'
+    documentation = '''
+This node transform the depthmaps from hololens recording into one coordinate system.
+'''
+
+    inputs = [
+        desc.File(
+            name="recordingDir",
+            label="Recording Folder",
+            description="HoloLens recording directory (images in pv/*.jpg, " \
+                "depth in long_throw_depth/*.pgm, + related .csv).",
+            value="",
+            uid=[0],
+        ),
+        desc.File(
+            name="UVfile",
+            label="UV decoding file",
+            description="Text file with transformation data from depthmap to " \
+                "depth camera coordinte system (uvdata.txt).",
+            value="",
+            uid=[0],
+        ),
+        desc.ChoiceParam(
+            name='verboseLevel',
+            label='Verbose Level',
+            description='''verbosity level (critical, error, warning, info, debug).''',
+            value='info',
+            values=['critical', 'error', 'warning', 'info', 'debug'],
+            exclusive=True,
+            uid=[],
+            ),
+        ]
+
+    outputs = [
+        desc.File(
+            name="output",
+            label="Output Folder",
+            description="",
+            value=desc.Node.internalFolder,
+            uid=[],
+            ),
+    ]
+
+    def processChunk(self, chunk):
+        try:
+            chunk.logManager.start(chunk.node.verboseLevel.value)
+            
+            if not chunk.node.recordingDir:
+                chunk.logger.warning('Nothing to process')
+                return
+            if not chunk.node.UVfile:
+                chunk.logger.warning('Missing depth decoding file (uvdata.txt)')
+                return
+            if not chunk.node.output.value:
+                return
```

Comment:
I am not sure what's the problem here. If I had to guess it would be probably because the user didn't input the output folder, it would be nice to let the user know, why it returned without doing anything (could we add some log message like in the previous two if conditions?)